WordPress/Sniffs/Theme/NonPrintableCheckSniff.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * WordPress Coding Standard.
+ *
+ * @package WPCS\WordPressCodingStandards
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+/**
+ * Restricts the use of non-printable characters in themes
+ *
+ * @link    https://make.wordpress.org/themes/handbook/review/required/theme-check-plugin/#info
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since   0.xx.0
+ */
+class WordPress_Sniffs_Theme_NonPrintableCheckSniff extends WordPress_Sniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'PHP',
+		'CSS',
+		'JS',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return ( array_merge( PHP_CodeSniffer_Tokens::$commentTokens, PHP_CodeSniffer_Tokens::$emptyTokens, PHP_CodeSniffer_Tokens::$stringTokens, array(
```

Comment:
I'm just wondering if there are still more tokens in which they could be _sneaked in_...

Regarding the build break for PHP7, if you changed the code within the `register()` method to the below, that should fix it:
```php
	$targets = array_merge(
		PHP_CodeSniffer_Tokens::$emptyTokens,
		PHP_CodeSniffer_Tokens::$heredocTokens,
		PHP_CodeSniffer_Tokens::$stringTokens
	);
	$targets[T_STRING]           = T_STRING;
	$targets[T_INLINE_HTML] = T_INLINE_HTML;

	if ( defined( 'T_BAD_CHARACTER' ) ) {
		$targets[T_BAD_CHARACTER] = T_BAD_CHARACTER;
	}

	return $targets;
```

Note: as we're defining a variable anyway, I've moved the extra tokens out of the `array_merge()`. `array_merge()` is expensive, while straight array assignments are not.
Also - for the extra ones, I've followed the pattern which the `PHP_CodeSniffer_Tokens` properties already use of having the token as both the key as well as the value in the array.